yt_dlp/extractor/mediaset.py

```diff
@@ -226,6 +226,7 @@ def _real_extract(self, url):
         first_e = None
```

Comment:
Although it doesn't help to acquire any media, it could be good to prioritise geo-blocking errors over missing formats, etc:
Here:
```
        first_e = geo_e = None
```
At l.240 below, after `first_e = e` (and `GeoRestrictedError` has to be imported):
```
                if not geo_e and isinstance(e, GeoRestrictedError):
                    geo_e = e
```
At l.253 below, instead of `raise first_e`:
```
                raise geo_e or first_e
```
So if no formats can be extracted it's really because formats were blocked from your region rather than because one of the API calls had no result.

Reply:
ok, done.